api/app/auto_spatial_advisory/process_fuel_type_area.py

```diff
@@ -0,0 +1,225 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_warped_fuel_type_s3_key(bucket):
+    """
+    Returns the key to the fuel type layer that has been reprojected to the Lambert Conformal Conic spatial reference and
+    transformed to match the extent and spatial reference of hfi files output by sfms.
+    """
+
+    # The filename in our object store, prepended with "vsis3" - which tells GDAL to use
+    # it's S3 virtual file system driver to read the file.
+    # https://gdal.org/user/virtual_file_systems.html
+    key = f"/vsis3/{bucket}/ftl/fuel_types_lambert_2000.tif"
+    return key
+
+
+def classify_by_threshold(source_data: np.array, threshold: int):
+    """
+    Classifies the provided 2-d array based on the provided threshold. When the threshold is 1, all cells with an hfi value
+    in the range of 4k - 10k are assigned a value of 1 and other cells are assigned a value of 0. When the threshold is 2,
+    all cells with an hfi greater than 10k are assigned a value of 1 and other cells are assigned a value of 0.
+
+    :param source_data: A 2-d array with values representing fuel types.
+    :param threshold: The current threshold being processed, 1 = 4k-10k, 2 = > 10k.
+    """
+    if threshold == 1:
+        # advisory
+        classified = np.where(source_data < 4000, 0, source_data)
+        classified = np.where((classified >= 4000) & (classified < 10000), 1, classified)
+        classified = np.where(classified >= 10000, 0, classified)
+    else:
+        # warning
+        classified = np.where(source_data < 10000, 0, source_data)
+        classified = np.where(classified >= 10000, 1, classified)
+    return classified
+
+
+async def calculate_fuel_type_area_by_shape(session: AsyncSession, temp_dir: str, source_path: str, threshold, run_parameters_id: int, fuel_types: list[SFMSFuelType]):
+    """
+    Process masked fuel type layer with each advisory shape (eg fire zone unit).
+
+    :param temp_dir: A temporary location for storing intermediate files.
+    :param source_path: Path to the masked fuel type layer.
+    :param threshold: The current threshold being processed, 1 = 4k-10k, 2 = > 10k.
+    :param run_parameters_id: The RunParameter object id associated with the run_type, for_date and run_datetime of interest.
+    :param fuel_types: A list of fuel types used in the sfms system.
+    """
+    stmt = text("SELECT id, source_identifier FROM advisory_shapes;")
+    result = await session.execute(stmt)
+    rows = result.all()
+    for row in rows:
+        shape_fuel_type_path = intersect_raster_by_advisory_shape(threshold, row[0], row[1], source_path, temp_dir)
+        fuel_type_areas = calculate_fuel_type_areas(shape_fuel_type_path, fuel_types)
+        await store_advisory_fuel_stats(session, fuel_type_areas, threshold, run_parameters_id, row[0])
+
+
+async def store_advisory_fuel_stats(session: AsyncSession, fuel_type_areas: dict, threshold: int, run_parameters_id: int, advisory_shape_id: int):
```

Comment:
I think this can go in `db/crud/auto_spatial_advisory.py` module since it's creating.

Reply:
Good call. Done!